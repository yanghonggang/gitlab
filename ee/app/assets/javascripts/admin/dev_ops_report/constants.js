import { s__ } from '~/locale';

export const DEVOPS_ADOPTION_STRINGS = {
  emptyState: {
    title: s__('DevopsAdoption|Add a segment to get started'),
    description: s__(
      'DevopsAdoption|DevOps adoption uses segments to track adoption across key features. Segments are a way to track multiple related projects and groups at once. For example, you could create a segment for the engineering department or a particular product team.',
    ),
    button: s__('DevopsAdoption|Add new segment'),
  },
};
