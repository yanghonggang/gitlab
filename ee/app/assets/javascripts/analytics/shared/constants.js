import { masks } from 'dateformat';

const { isoDate, mediumDate } = masks;

export const dateFormats = {
  isoDate,
  defaultDate: mediumDate,
  defaultDateTime: 'mmm d, yyyy h:MMtt',
};

export const scatterChartLineProps = {
  default: {
    type: 'line',
    showSymbol: false,
    // By default zlevel is 2 for all series types.
    // By increasing the zlevel to 3 we make sure that the trendline gets drawn in front of the dots in the chart.
    zlevel: 3,
  },
};

export const DATE_RANGE_LIMIT = 180;

export const OFFSET_DATE_BY_ONE = 1;

export const NO_DRAG_CLASS = 'no-drag';

export const DATA_REFETCH_DELAY = 250;
