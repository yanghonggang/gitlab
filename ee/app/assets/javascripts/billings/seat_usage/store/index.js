import * as actions from './actions';
import mutations from './mutations';
import state from './state';

export default (initState = {}) => ({
  actions,
  mutations,
  state: state(initState),
});
