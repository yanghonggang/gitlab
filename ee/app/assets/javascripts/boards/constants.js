export const DRAGGABLE_TAG = 'div';

/* eslint-disable @gitlab/require-i18n-strings */
export const EpicFilterType = {
  any: 'Any',
  none: 'None',
};

export const IterationFilterType = {
  any: 'Any',
  none: 'None',
};

export const GroupByParamType = {
  epic: 'epic',
};

export default {
  DRAGGABLE_TAG,
  EpicFilterType,
};
