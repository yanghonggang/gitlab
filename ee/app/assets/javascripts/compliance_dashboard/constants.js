export const PRESENTABLE_APPROVERS_LIMIT = 2;

export const COMPLIANCE_TAB_COOKIE_KEY = 'compliance_dashboard_tabs';

export const INPUT_DEBOUNCE = 500;

export const CUSTODY_REPORT_PARAMETER = 'commit_sha';
