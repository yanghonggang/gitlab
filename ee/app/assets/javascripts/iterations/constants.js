export const Namespace = {
  Group: 'group',
  Project: 'project',
};
