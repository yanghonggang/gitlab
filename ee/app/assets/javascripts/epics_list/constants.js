import { __ } from '~/locale';
import { AvailableSortOptions } from '~/issuable_list/constants';

export const EpicsSortOptions = [
  {
    id: AvailableSortOptions.length + 1,
    title: __('Start date'),
    sortDirection: {
      descending: 'start_date_desc',
      ascending: 'start_date_asc',
    },
  },
  {
    id: AvailableSortOptions.length + 2,
    title: __('Due date'),
    sortDirection: {
      descending: 'end_date_desc',
      ascending: 'end_date_asc',
    },
  },
];

export const FilterTokenOperators = [
  { value: '=', description: __('is'), default: 'true' },
  // { value: '!=', description: __('is not') },
];
