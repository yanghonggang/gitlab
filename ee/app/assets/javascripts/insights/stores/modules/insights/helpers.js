import { CHART_TYPES } from 'ee/insights/constants';
import { __ } from '~/locale';

const getAxisTitle = label => {
  switch (label) {
    case 'day':
      return __('Days');
    case 'week':
      return __('Weeks');
    case 'month':
      return __('Months');
    case 'issue':
      return __('Issues');
    case 'merge_request':
      return __('Merge requests');
    default:
      return '';
  }
};

export const transformChartDataForGlCharts = (
  { type, query: { group_by, issuable_type } },
  { labels, datasets },
) => {
  const formattedData = {
    xAxisTitle: getAxisTitle(group_by),
    yAxisTitle: getAxisTitle(issuable_type),
    labels,
    datasets: [],
    seriesNames: [],
  };

  switch (type) {
    case CHART_TYPES.BAR:
      formattedData.datasets = [
        {
          name: 'all',
          data: labels.map((label, i) => [label, datasets[0].data[i]]),
        },
      ];
      break;
    case CHART_TYPES.STACKED_BAR:
      formattedData.datasets.push(
        ...datasets.map(dataset => ({
          name: dataset.label,
          data: dataset.data,
        })),
      );
      break;
    case CHART_TYPES.LINE:
      formattedData.datasets.push(
        ...datasets.map(dataset => ({
          name: dataset.label,
          data: labels.map((label, i) => [label, dataset.data[i]]),
        })),
      );

      break;
    default:
      formattedData.datasets = { all: labels.map((label, i) => [label, datasets[0].data[i]]) };
  }
  return formattedData;
};

export default {
  transformChartDataForGlCharts,
};
