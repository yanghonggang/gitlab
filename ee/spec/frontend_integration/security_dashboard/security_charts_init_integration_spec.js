import initSecurityCharts from 'ee/security_dashboard/security_charts_init';
import { DASHBOARD_TYPES } from 'ee/security_dashboard/store/constants';
import { TEST_HOST } from 'jest/helpers/test_constants';

const EMPTY_DIV = document.createElement('div');

const TEST_DATASET = {
  link: '/test/link',
  svgPath: '/test/no_changes_state.svg',
  dashboardDocumentation: '/test/dashboard_page',
  emptyStateSvgPath: '/test/empty_state.svg',
};

describe('Security Charts', () => {
  let vm;
  let root;

  beforeEach(() => {
    root = document.createElement('div');
    document.body.appendChild(root);

    global.jsdom.reconfigure({
      url: `${TEST_HOST}/-/security/dashboard`,
    });
  });

  afterEach(() => {
    if (vm) {
      vm.$destroy();
    }
    vm = null;
    root.remove();
  });

  const createComponent = ({ data, type }) => {
    const el = document.createElement('div');
    Object.assign(el.dataset, { ...TEST_DATASET, ...data });
    root.appendChild(el);
    vm = initSecurityCharts(el, type);
  };

  const createEmptyComponent = () => {
    vm = initSecurityCharts(null, null);
  };

  describe('default states', () => {
    it('sets up group-level', () => {
      createComponent({ data: { groupFullPath: '/test/' }, type: DASHBOARD_TYPES.GROUP });

      expect(root).toMatchSnapshot();
    });

    it('sets up instance-level', () => {
      createComponent({
        data: { instanceDashboardSettingsPath: '/instance/settings_page' },
        type: DASHBOARD_TYPES.INSTANCE,
      });

      expect(root).toMatchSnapshot();
    });
  });

  describe('error states', () => {
    it('does not have an element', () => {
      createEmptyComponent();

      expect(root).toStrictEqual(EMPTY_DIV);
    });

    it('has unavailable pages', () => {
      createComponent({ data: { isUnavailable: true } });

      expect(root).toMatchSnapshot();
    });
  });
});
