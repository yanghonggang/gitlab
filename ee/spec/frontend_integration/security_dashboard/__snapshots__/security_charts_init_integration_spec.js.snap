// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security Charts default states sets up group-level 1`] = `
<div>
  <div
    data-testid="security-charts-layout"
  >
    <h2>
      Security Dashboard
    </h2>
    <div
      class="gl-spinner-container gl-mt-6"
    >
      <span
        aria-label="Loading"
        class="align-text-bottom gl-spinner gl-spinner-dark gl-spinner-lg"
      />
    </div>
    <div
      class="security-charts gl-display-flex gl-flex-wrap"
    />
  </div>
</div>
`;

exports[`Security Charts default states sets up instance-level 1`] = `
<div>
  <div
    data-testid="security-charts-layout"
  >
    <h2>
      Security Dashboard
    </h2>
    <div
      class="gl-spinner-container gl-mt-6"
    >
      <span
        aria-label="Loading"
        class="align-text-bottom gl-spinner gl-spinner-dark gl-spinner-lg"
      />
    </div>
    <div
      class="security-charts gl-display-flex gl-flex-wrap"
    />
  </div>
</div>
`;

exports[`Security Charts error states has unavailable pages 1`] = `
<div>
  <section
    class="row empty-state text-center"
  >
    <div
      class="col-12"
    >
      <div
        class="svg-250 svg-content"
      >
        <img
          alt="Oops, something doesn't seem right."
          class="gl-max-w-full"
          src="/test/empty_state.svg"
        />
      </div>
    </div>
    <div
      class="col-12"
    >
      <div
        class="text-content gl-mx-auto gl-my-0 gl-p-5"
      >
        <h1
          class="h4"
        >
          Oops, something doesn't seem right.
        </h1>
        <p>
          Either you don't have permission to view this dashboard or the dashboard has not been setup. Please check your permission settings with your administrator or check your dashboard configurations to proceed.
        </p>
        <div>
          <a
            class="btn btn-success btn-md gl-button"
            href="/test/dashboard_page"
          >
            <span
              class="gl-button-text"
            >
              Learn more about setting up your dashboard
            </span>
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
